src/cartridges/int_adyen_SFRA/cartridge/scripts/hooks/payment/processor/middlewares/handle.js

```diff
@@ -1,52 +1,64 @@
 const Transaction = require('dw/system/Transaction');
+const PaymentInstrument = require('dw/order/PaymentInstrument');
 const AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
 const collections = require('*/cartridge/scripts/util/collections');
 const constants = require('*/cartridge/adyenConstants/constants');
 
+function removeAllPaymentInstruments(currentBasket) {
+  collections.forEach(currentBasket.getPaymentInstruments(), (item) => {
+    currentBasket.removePaymentInstrument(item);
+  });
+}
+
+function convertToSfccCardType(paymentInformation, paymentInstrument) {
+  const sfccCardType = !paymentInformation.creditCardToken
+    ? AdyenHelper.getSFCCCardType(paymentInformation.cardType)
+    : paymentInformation.cardType;
+
+  paymentInstrument.setCreditCardNumber(paymentInformation.cardNumber);
+  paymentInstrument.setCreditCardType(sfccCardType);
+
+  paymentInstrument.custom.adyenPaymentMethod = sfccCardType;
+
+  if (paymentInformation.creditCardToken) {
+    paymentInstrument.setCreditCardExpirationMonth(
+      paymentInformation.expirationMonth,
+    );
+    paymentInstrument.setCreditCardExpirationYear(
+      paymentInformation.expirationYear,
+    );
+    paymentInstrument.setCreditCardToken(paymentInformation.creditCardToken);
+  }
+}
+
 function handle(basket, paymentInformation) {
   const currentBasket = basket;
   const cardErrors = {};
   const serverErrors = [];
 
   Transaction.wrap(() => {
-    collections.forEach(currentBasket.getPaymentInstruments(), (item) => {
-      currentBasket.removePaymentInstrument(item);
-    });
+    removeAllPaymentInstruments(currentBasket);
+
+    const paymentInstrumentType = paymentInformation.isCreditCard
+      ? PaymentInstrument.METHOD_CREDIT_CARD
+      : constants.METHOD_ADYEN_COMPONENT;
     const paymentInstrument = currentBasket.createPaymentInstrument(
```

Comment:
let's add a unit test for this to make sure it chooses the right one when creating a payment instrument

Reply:
Test added for checking payment type.